<details><summary>Linter check found the following problems:</summary>
Traceback (most recent call last):
  File "/opt/conda/envs/linter/lib/python3.13/site-packages/anaconda_linter/lint/__init__.py", line 840, in lint_file
    recipe = _recipe.Recipe.from_file(
        recipe_fname=str(meta_yaml),
    ...<2 lines>...
        renderer=RendererType.RUAMEL,
    )
  File "/opt/conda/envs/linter/lib/python3.13/site-packages/percy/render/recipe.py", line 245, in from_file
    raise exc from exc
  File "/opt/conda/envs/linter/lib/python3.13/site-packages/percy/render/recipe.py", line 236, in from_file
    recipe._load_from_string(text.read())
    ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "/opt/conda/envs/linter/lib/python3.13/site-packages/percy/render/recipe.py", line 172, in _load_from_string
    self.render()
    ~~~~~~~~~~~^^
  File "/opt/conda/envs/linter/lib/python3.13/site-packages/percy/render/recipe.py", line 297, in render
    self.meta = renderer_utils.render(self.recipe_file, self.dump(), self.selector_dict, self.renderer)
                ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/linter/lib/python3.13/site-packages/percy/render/_renderer.py", line 260, in render
    raise JinjaRenderFailure(recipe_file, message=exc.message) from exc
percy.render.exceptions.JinjaRenderFailure: (PosixPath('/tmp/abs_81st1794vo/clone/recipe/meta.yaml'), "'lz4' is undefined")

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/conda/envs/linter/lib/python3.13/site-packages/anaconda_linter/run.py", line 209, in main
    return_code, report = execute_linter(
                          ~~~~~~~~~~~~~~^
        recipe=args.recipe,
        ^^^^^^^^^^^^^^^^^^^
    ...<6 lines>...
        verbose_flag=args.verbose,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/conda/envs/linter/lib/python3.13/site-packages/anaconda_linter/run.py", line 181, in execute_linter
    result = linter.lint(recipes, subdir, variant_config_files, exclusive_config_files, fix_flag)
  File "/opt/conda/envs/linter/lib/python3.13/site-packages/anaconda_linter/lint/__init__.py", line 742, in lint
    msgs = self.lint_file(
        recipe_name,
    ...<3 lines>...
        fix=fix,
    )
  File "/opt/conda/envs/linter/lib/python3.13/site-packages/anaconda_linter/lint/__init__.py", line 857, in lint_file
    recipe = _recipe.Recipe(recipe_name)
  File "/opt/conda/envs/linter/lib/python3.13/site-packages/percy/render/recipe.py", line 109, in __init__
    self.recipe_dir = recipe_file.parent
                      ^^^^^^^^^^^^^^^^^^
AttributeError: 'str' object has no attribute 'parent'

ERROR conda.cli.main_run:execute(127): `conda run conda-lint /tmp/abs_81st1794vo/clone` failed. (See above for error)
</details>

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -59,7 +59,24 @@ requirements:
 
 test:
   imports:
-    - pysam
+    - pysam.config
+    - pysam.libcalignedsegment
+    - pysam.libcalignmentfile
+    - pysam.libcbcf
+    - pysam.libcbcftools
+    - pysam.libcbgzf
+    - pysam.libcfaidx
+    - pysam.libchtslib
+    - pysam.libcsamfile
+    - pysam.libcsamtools
+    - pysam.libctabix
+    - pysam.libctabixproxies
+    - pysam.libcutils
+    - pysam.libcvcf
+    - pysam.Pileup
+    - pysam.samtools
+    - pysam.utils
+    - pysam.version
   commands:
     - pip check
     - python -c "from importlib.metadata import version; assert(version('{{ name }}')=='{{ version }}')"

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -37,14 +37,14 @@ requirements:
     - wheel
     - cython >=3,<4
     # These are required for the vendored htslib.
-    - libcurl 8
+    - libcurl {{ libcurl }}
     - bzip2 {{ bzip2 }}
     - xz {{ xz }}
-    - libdeflate 1.22
+    - libdeflate {{ libdeflate }}
     # Only required for libcrypto.
     # On macOS, links to libSystem
     - openssl  {{ openssl }}  # [linux]
-    - lz4-c 1.9
+    - lz4-c {{ lz4_c }}
     # These are required for the vendored htslib, bcftools and samtools.
     - zlib {{ zlib }}
   run:

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -19,6 +19,8 @@ build:
   script_env:
     # This is the default, but it's preferable to be explicit.
     - HTSLIB_MODE=shared
+    - SHLIB_EXT=.so # [linux]
+    - SHLIB_EXT=.dylib # [osx]
   script:
     # Delete samtools/lz4 just to make sure they are not used/compiled by accident.
     - rm -rf samtools/lz4
@@ -98,7 +100,4 @@ about:
     such as those routinely created by next-generation sequencing methods.
   dev_url: https://github.com/pysam-developers/pysam
   doc_url: https://pysam.readthedocs.io
-
-extra:
-  skip-lints:
-    - patch_unnecessary
+  

--- a/recipe/patches/0001-Unvendor-deps.patch
+++ b/recipe/patches/0001-Unvendor-deps.patch
@@ -1,5 +1,19 @@
---- setup.py	2025-08-14 09:28:27
-+++ setup_new.py	2025-08-14 09:43:06
+From f40f36cae9cab99c5544c6707f18724cf1fc9632 Mon Sep 17 00:00:00 2001
+From: Jean-Christophe Morin <jcmorin@anaconda.com>
+Date: Fri, 2 Feb 2024 13:51:49 -0500
+Subject: [PATCH] Unvendor lz4 and compile a dynalic htslib instead of static.
+
+Note that we don't unvendor samtools and bcftools because they are modified from upstream
+and we can't unvendor them. As for htslib, it was considered unnecessary to unvendor
+since its version is tightly coupled with pysam, bcftools and samtools.
+---
+ setup.py | 20 +++++++++++-------
+ 1 file changed, 11 insertions(+), 7 deletions(-)
+
+diff --git a/setup.py b/setup.py
+index 455b42a..df88487 100644
+--- a/setup.py
++++ b/setup.py
 @@ -500,6 +500,9 @@
      # all object files are pulled into the link, even those not used by htslib itself.
      htslib_objects = [os.path.join("htslib", x)

